EIPS/eip-3224.md

```diff
@@ -0,0 +1,465 @@
+only addresses honest contracts that want to make their user's
+life better. We believe that this is a reasonable security model,
+as transaction descriptions can be audited at the same time as
+contract code, allowing auditors and code reviewers to check that
+transaction descriptions are accurate as part of their review.
+
+
+## Specification
+
+The **description string** and **described data** are generated
+simultaneously by evaluating **describer** bytecode, passed
+**describedInput** as the calldata with the encoded method:
+
+```solidity
+function eipXXXDescribe(bytes inputs) view returns (string description, bytes data);
+```
+
+The method MUST be executable in a static context, i.e. any
+side effects (logX, sstore, etc., (including through indirect
+calls) will be ignored.
+
+During evaluation, the `ADDRESS` (i.e. `to`), `CALLER`
+(i.e. `from`), `VALUE`, and `GASPRICE` must be the same as the
+values for the transaction being described, so that the
+code generating the description can rely on them. For signing
+**described messages**, `VALUE` should always be 0.
+
+When executing the bytecode, best efforts should be made to
+ensure `BLOCKHASH`, `NUMBER`, `TIMESTAMP` and `DIFFICULTY`
+match the `"latest"` block. The `COINBASE` should be the zero
+address.
+
+If evaluating the function results in a revert, and the return
+data is of a length that is congruent to `4 mod 32`, and the
+first 4 bytes represent the selector `Error(string)`, the
+reason string should be displayed to the user. If evaluation
+reverts otherwise, then a generic evaluation error should be shown.
+In either case, the signing is aborted.
+
+
+### New JSON-RPC Methods
+
+Clients which manage private keys should expose additional
+methods for interacting with the related accounts.
+
+If an user interface is not present or expected for any other
+account-based operations, the description strings should be
+ignored and the described data used directly.
+
+These JSON-RPC methods will also be implemented in standard
+Ethereum libraries, so the JSON-RPC description is meant more
+of a canonical way to describe them.
+
+
+### Signing Described Messages
+
+```solidity
+eth_signDescribedMessage(address, describer, describerInput)
+// Result: {
+//   description: "text/plain;Hello World",
+//   data: "0x...", // described data
+//   signature: "0x..."
+// }
+```
+
+Compute the **description string** and **described data** by
+evaluating the call to **describer**, with the
+**describerInput** passed to the ABI encoded call to
+`eipXXXDescription(bytes)`. The `VALUE` during execution must
+be 0.
+
+If the wallet contains a user interface for accepting or
+denying signing a message, it should present the description
+string to the user. Optionally, a wallet may wish to
+additionally provide a way to examine the described data.
+
+If accepted, the computed **described data** is signed
+according to [EIP-191](./eip-191.md), with the *version
```

Comment:
Note: EIP-191 is still just a draft, so this EIP will get blocked by that unless you can move it forward.

Reply:
Interesting... @Arachnid, is there any reason not to move this forward?

Reply:
I think for now this is fine, but will need to be addressed later. But we certainly don't want to define some new standard that's not already in the works for something like this. EIP-191 is already widely used.